File: drools-spring-v2-app/src/main/java/drools/spring/example/facts/Administrator.java
package drools.spring.example.facts;

import javax.persistence.Entity;
import javax.persistence.PrimaryKeyJoinColumn;

@Entity
@PrimaryKeyJoinColumn(name = "id")
public class Administrator extends User{
	
	public Administrator() {
		super();
	}

	public Administrator(Long id, String username, String password, String name, String surname) {
		super(id, username, password, name, surname, ROLE.ADMIN);
	}
	
}

File: drools-spring-v2-app/src/main/frontend/src/app/model/administrator.ts
import { ROLE } from '../model/role.enum';
import { User } from '../model/user';

export class Administrator extends User{
    constructor(username : string, passsword : string, name : string, surname : string, email : string){
        super(username, passsword, name, surname, email, ROLE.ADMIN);
    }
}
